### ISSUE TITLE
new generic api for Intro/Credit

### ISSUE THREAD

**Issue by jumoog** [jumoog/intro-skipper#193]
I had planned to change this as well. I could already provide the API. What would make sense from your point of view?

```json
{
    "Introduction": {
        "EpisodeId": "1a103d3ac9cd3be7221655f856f460d4",
        "Valid": true,
        "Start": 112.16279999999999,
        "End": 130.5898,
        "ShowSkipPromptAt": 112.16279999999999,
        "HideSkipPromptAt": 129.5898
    },
    "Credits": {
        "EpisodeId": "1a103d3ac9cd3be7221655f856f460d4",
        "Valid": true,
        "Start": 1271.8206,
        "End": 1290,
        "ShowSkipPromptAt": 1271.8206,
        "HideSkipPromptAt": 1289
    }
}
```

**Comment by AbandonedCart** [comment]
Having tried findroid myself, I honestly have no idea what that last post you referenced is going on about. The app is already compatible with the plugin and almost all of the code in that PR is related to adding credits (which haven't changed from the original plugin) and fixing bugs with their current implementation.

That said, I don't think changing the API right now is a good idea. You have jellyfin trying to implement their own version of an API, you have a lot of people still basing their code on the original plugin API, and you have almost weekly updates to the platform. With how many bugs are already being reported is another major change really going to improve things?

**Comment by jumoog** [comment]
Of course, the old one stays and another one is added. I understood him to mean that he would like a better API

**Comment by jumoog** [comment]
Of course, the old one stays and a new one is added. I understood him to mean that he wanted a better API.

**Comment by AbandonedCart** [comment]
Oh ok. It was actually about how they can optimize their use of it.
